src/providers/useReducerProvider.js

```diff
@@ -0,0 +1,54 @@
+import { createContext, useContext, useReducer, useState } from "react";
+
+const ReducerContext = createContext();
+
+export function ReducerProvider({ children }) {
+  const [isSideNav, setIsSideNav] = useState(false);
+  const [state, dispatch] = useReducer(reducer, {
+    history: [],
+    watchLater: [],
+    likedVideos: [],
+    Data: [],
+    product: {},
+  });
+  function reducer(state, value) {
```

Comment:
Reducer function can be separated into a different file, and export it from there.